### ISSUE TITLE
[HOLD for payment 2023-05-04] [$1000]  Web - Chat - The frame is cut off if you use "Tab" key with message options

### ISSUE THREAD

**Issue by kbecciv** [Expensify/App#17791]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Access staging.new.expensify.com
2. Go to any chat
3. Click on message to show message options
4. Use the Tab key to navigate through the list of message options.
5. Select the option to delete the message

## Expected Result:
The corners of the frame are not cut.

## Actual Result:
If you use the Tab key with message options, the corners of the frame are cut off.

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome
- [ ] MacOS / Desktop

**Version Number:** 1.3.3.1

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

![Untitled (4)](https://user-images.githubusercontent.com/93399543/233676258-c3366379-833c-40a1-8a90-5a9c51450a50.png)

https://user-images.githubusercontent.com/93399543/233676261-9c1abcd4-89ec-494b-9180-516f2ed23301.mp4

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e902d4a196c392ab</li>
        <li>Upwork Job ID: 1649506039835144192</li>
        <li>Last Price Increase: 2023-04-21</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by abekkala** [comment]
was able to reproduce.

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~01e902d4a196c392ab

**Comment by MelvinBot** [comment]
Current assignee @abekkala is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @s77rt (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @stitesExpensify (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by Pujan92** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Focus element blue outline cut for last element in context menu

### What is the root cause of that problem?
Context menu has a border radius which is causing this issue where blue outline isn't fully visible.

### What changes do you think we should make in order to solve the problem?
To fix we can provide a padding of 4px/8px with `styles.pb2` to ContextMenuItem popover the same way we are providing for other popovers.

**Comment by MahmudjonToraqulov** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
Focus element blue outline cut for last element in context menu

### What is the root cause of that problem?
 It's parent element has `border-radius: 12px` and `overflow: hidden` but It has `border-radius: 0px`.  `overflow: hidden`  style doesn't let its bottom borders to look. 

### What changes do you think we should make in order to solve the problem?
 To solve this issue we can add `border-radius: 0 0 12px 12px` to the element.  You can get better in the video below.

### What alternative solutions did you explore? (Optional)
N/A

### Video
[screen-capture (25).webm](https://user-images.githubusercontent.com/91727683/233765415-32733595-fc1e-4f1e-aa4c-e76bd73be956.webm)

**Comment by s77rt** [comment]
@Pujan92 Thanks for the proposal. Your RCA is correct and the fix is about right but I think it can be improved. We should set vertical padding (top and bottom) instead of just bottom padding otherwise context menus with no reaction items will look weird e.g. the context menu that you get after right clicking on url/email. However setting vertical padding will make the main context menu a little empty at the top.

Can you please update your proposal and explain how you solve the extra empty space problem? I think this has to do with [emojiReactionBubble](https://github.com/Expensify/App/blob/4947508bedc2e901082d805c84498f5f394ef655/src/styles/styles.js#L3018-L3026) but please feel free to explore any other options you see fit.

**Comment by s77rt** [comment]
@MahmudjonToraqulov Thanks for the proposal. Your RCA is correct. The suggested fix is also valid but for consistency reasons we want to keep the blue outline in a rectangle shape and just add vertical padding to the container to avoid items getting cut as can been seen in the FAB.

![Screenshot from 2023-04-22 21-15-35](https://user-images.githubusercontent.com/16493223/233804566-7fe06bc6-ea7d-4595-bce9-3569c30fef18.png)

**Comment by jayeshmangwani** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
The frame border is cut from the first and last element in the popover context menu.

### What is the root cause of that problem?
We are using a border radius for the parent component, and the frame has outlined the square, so the first and last element's border will cut from the corners due to no padding to the first and last elements.

### What changes do you think we should make in order to solve the problem?
For consistency in the app (FAB button, Payment Option), We need to provide the `paddingVertical: 12(styles.pv3)` in the context menu container and all the other remaining popovers where we have not provided vertical padding, and we only have to provide the vertical padding if the screen is not narrow(!isSmallScreenWidth) because we are already passing the default paddingTop to the modal type `CONST.MODAL.MODAL_TYPE.BOTTOM_DOCKED(where screen width will be small)`
https://github.com/Expensify/App/blob/4947508bedc2e901082d805c84498f5f394ef655/src/styles/getModalStyles/getBaseModalStyles.js#L170


#### I have reported this same bug with the possible solution in [Slack](https://expensify.slack.com/archives/C049HHMV9SM/p1675804299091569) here but at that time I wasn't sure if we want to fix accessibility bugs

Able to find this padding bug in 2 places, one at the Context menu and another at the Call option from the chat Header

1. For the context menu, we need to pass the pv3 to the parent View of the Popover when isSmallScreenWidth value false

https://github.com/Expensify/App/blob/4947508bedc2e901082d805c84498f5f394ef655/src/styles/getReportActionContextMenuStyles.js#L30-L32

Here we need to create a new prop name, `isSmallScreenWidth` and if the isSmallScreenWidth value is false, then pass the styles.pv3 to `bigWrapperStyle` 
For the emoji reaction part, I would suggest that we should remove the Top padding from the reaction container and provide only padding-bottom because we are passing the padding-top(pv3) to the main container, and emoji reaction is the first element in the context menu so it always has the top padding and another benefit is that is if we hide the quick emoji reaction view then default top and bottom padding will be as it is, as it is not depending on emoji reaction 

**The result after this change**
<img width="536" alt="Screenshot 2023-04-23 at 3 09 40 AM" src="https://user-images.githubusercontent.com/35371050/233808557-55e42eec-01af-4fac-a65a-56290a35007a.png">

![Simulator Screen Shot - iPhone 14 - 2023-04-23 at 03 09 52](https://user-images.githubusercontent.com/35371050/233808558-20879714-c8e3-4496-9050-fc50c38bccb4.png)

2. For the video and call option model, the solution is straightforward
wrap the PopOver child in the View and pass the pv3 if the isSmallScreenWidth value is false

https://github.com/Expensify/App/blob/4947508bedc2e901082d805c84498f5f394ef655/src/components/VideoChatButtonAndMenu/BaseVideoChatButtonAndMenu.js#L134-L142

Wrap this in this View 
```
<View style={this.props.isSmallScreenWidth ? {} : styles.pv3}>
{_.map(this.menuItemData, ({icon, text, onPress}) => (
    <MenuItem
        wrapperStyle={styles.mr3}
        key={text}
        icon={icon}
        title={text}
        onPress={onPress}
    />
))}
</View>
```
**The result after this change**
<img width="362" alt="Screenshot 2023-04-23 at 3 09 14 AM" src="https://user-images.githubusercontent.com/35371050/233808864-3071fd2e-c4cf-4b82-858a-cc6a43a615e3.png">


### What alternative solutions did you explore? (Optional)
- none

**Comment by s77rt** [comment]
@jayeshmangwani Thanks for the proposal. Your RCA is correct and I like your suggested solution. The reaction part feels like a workaround but I won't block on that given that your solution is actually a complete solution to fix this issue. We will handle the reaction part separately. 

:ribbon: :eyes: :ribbon: C+ reviewed

cc @stitesExpensify 

___

@stitesExpensify Based on the https://expensify.slack.com/archives/C02NK2DQWUX/p1680798564574399?thread_ts=1678889255.458839&cid=C02NK2DQWUX process, would you please create an issue for https://github.com/Expensify/App/pull/17850 or reopen https://github.com/Expensify/App/issues/15808.

**Comment by stitesExpensify** [comment]
CC @shawnborton this solution seems good to me, I just want to confirm you think it looks okay from a design perspective 
https://github.com/Expensify/App/issues/17791#issuecomment-1518880870

**Comment by s77rt** [comment]
@stitesExpensify @jayeshmangwani already link to an old discussion https://expensify.slack.com/archives/C049HHMV9SM/p1675804299091569

![Screenshot from 2023-04-24 17-15-17](https://user-images.githubusercontent.com/16493223/234055690-05cc158d-72c3-420c-9212-1ef8ee77d80c.png)

**Comment by stitesExpensify** [comment]
Ah thank you very much, I missed that. No input necessary Shawn! 

Assigning @jayeshmangwani for their proposal

**Comment by MelvinBot** [comment]
📣 @jayeshmangwani You have been assigned to this job by @stitesExpensify!
Please apply to this job in Upwork and leave a comment on the Github issue letting us know when we can expect a PR to be ready for review 🧑‍💻
Keep in mind: [Code of Conduct](https://github.com/Expensify/App/blob/main/CODE_OF_CONDUCT.md) | [Contributing](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md) 📖

**Comment by shawnborton** [comment]
Sounds good :)

**Comment by jayeshmangwani** [comment]
[PR](https://github.com/Expensify/App/pull/17909) raised
@s77rt @stitesExpensify

**Comment by MelvinBot** [comment]
`Reviewing` label has been removed, please complete the "BugZero Checklist".

**Comment by MelvinBot** [comment]
The solution for this issue has been :rocket: deployed to production :rocket: in version 1.3.6-0 and is now subject to a 7-day regression period :calendar:. Here is the list of pull requests that resolve this issue:

- https://github.com/Expensify/App/pull/17909


If no regressions arise, payment will be issued on 2023-05-04. :confetti_ball:

After the hold period is over and BZ checklist items are completed, please complete any of the applicable payments for this issue, and check them off once done.
- [ ] External issue reporter
- [ ] Contributor that fixed the issue
- [ ] Contributor+ that helped on the issue and/or PR

As a reminder, here are the bonuses/penalties that should be applied for any External issue:
- Merged PR within 3 business days of assignment - 50% **bonus**
- Merged PR more than 9 business days after assignment - 50% **penalty**

**Comment by MelvinBot** [comment]
**BugZero Checklist:** The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

- [ ] [@s77rt] The PR that introduced the bug has been identified. Link to the PR:
- [ ] [@s77rt] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment:
- [ ] [@s77rt] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion:
- [ ] [@s77rt] Determine if we should create a regression test for this bug.
- [ ] [@s77rt] If we decide to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the same bug will not reach production again.
- [ ] [@abekkala] Link the GH issue for creating/updating the regression test once above steps have been agreed upon:

**Comment by s77rt** [comment]
- The PR that introduced the bug has been identified: I think this existed since day one, modals always had a border radius https://github.com/Expensify/App/commit/75eac9e0967539a321fb923ddab645aa410b4131 and that is even before the implementation of the context menu. So I guess the most correct answer on the offending PR is https://github.com/Expensify/App/pull/1678
- The offending PR has been commented on: I can't comment on that PR. @stitesExpensify Do we really need to point this out?
- A discussion in #expensify-bugs has been started: We already have a similar discussion https://expensify.slack.com/archives/C049HHMV9SM/p1681981488202219
- Determine if we should create a regression test for this bug: Yes

___

**Regression Test Proposal**


1. Open App
2. Go to any chat
3. Right Click / Long Press on any message to open the context menu
4. Keep pressing the TAB key to navigate through the options
5. Verify a blue frame appears on each option
6. Verify the corners of the frame are not cut
7. Close the context menu
8. Press the call button from the chat header
9. Keep pressing the TAB key to navigate through the options
10. Verify a blue frame appears on each option
11. Verify the corners of the frame are not cut

**Comment by abekkala** [comment]
Payments to be made on May 04:
- Issue reported by: Applause - Internal Team [no payout]
- Selected Proposal for fix: @jayeshmangwani [$1000] + [$500 Bonus]
- C+ Review: @s77rt  [$1000] + [$500 Bonus]


Apr 24 - @jayeshmangwani was assigned to this job
Apr 24 - PR merged https://github.com/Expensify/App/pull/17909
Apr 26 - Deployed to staging
Apr 26 - Deployed to production
no regressions: 50% bonus payout 🎉

**Comment by jayeshmangwani** [comment]
@abekkala I accepted the offer to Upwork, and I want to confirm if I am eligible for reporting bonus here, as I raised this issue in [Slack](https://expensify.slack.com/archives/C049HHMV9SM/p1675804299091569) few months back

**Comment by abekkala** [comment]
@jayeshmangwani I'll add the $250 to your payout

**Comment by abekkala** [comment]
@jayeshmangwani paid and contract ended! 🎉 Thanks!

**Comment by abekkala** [comment]
@s77rt you've been paid and contract ended! 🎉 Thank you!
